view/frontend/web/js/view/payment/method-renderer/base.js

```diff
@@ -0,0 +1,409 @@
+define([
+        'underscore',
+        'jquery',
+        'ko',
+        'uiComponent',
+        'acquiredLoader',
+        'mage/translate',
+        'Magento_Checkout/js/model/full-screen-loader',
+        'mage/storage',
+        'Magento_Checkout/js/model/url-builder',
+        'Magento_Ui/js/model/messageList',
+        'Magento_Checkout/js/action/redirect-on-success'
+], function (
+    _,
+    $,
+    ko,
+    Component,
+    acquiredLoader,
+    $t,
+    fullScreenLoader,
+    storage,
+    urlBuilder,
+    messageList,
+    redirectOnSuccessAction,
+) {
+
+    return Component.extend({
+
+        defaults: {
+            acquired: null,
+            code: 'acquired_card',
+            active: false,
+            nonce: null,
+            acquiredComponent: null,
+            sessionId: null,
+            placeholder: '#acquired-payments-card-component',
+            /**
+             * Additional payment data
+             *
+             * {Object}
+             */
+            orderId: '',
+            transactionId: '',
+            timestamp: '',
+            imports: {
+                onActiveChange: 'active',
+            },
+            isPlaceOrderActionAllowed: ko,
+        },
+
+        generateNonce: function () {
+            const generateHash = function (input) {
+                let hash = 0;
+                for (let i = 0; i < input.length; i++) {
+                    hash = Math.imul(31, hash) + input.charCodeAt(i) | 0;
+                }
+                return ('00000000' + (hash >>> 0).toString(16)).slice(-8);
+            };
+
+            return generateHash(Math.random().toString(36).substring(2) + Date.now());
```

Comment:
Does this nonce need to be cryptographically secure? If so we shouldn't use `Math.random()`.

- [ ] Validate use of `Math.random()`

Reply:
Hi Adam, well in general it is there only to validate the state of payment intent of sorts, if cryptographical security is of concern I suggest the following:

```
let random = () => {
   let crypto = window.crypto || window.cryptoMs;
   if( crypto ) {
     return crypto.getRandomValues(new Uint32Array(1))[0]/2**32;
   }
   return Math.random();
};
let seed = random().toString(36).substring(2) + Date.now();
```

If browser supports window.crypto or window.cryptoMs use that to generate the random values.

Reply:
@adamcox-acquired can you let me know if the above works or not?

Reply:
If there's no need for it to be cryptographically secure then I don't think we need to make a change. Thanks for confirming.